cl_dll/hud_customtimer.cpp

```diff
@@ -0,0 +1,61 @@
+#include "hud.h"
+#include "cl_util.h"
+
+DECLARE_COMMAND(m_CustomTimer, CustomTimer);
+
+int CHudCustomTimer::Init()
+{
+	HOOK_COMMAND("customtimer", CustomTimer);
+
+	m_iFlags = 0;
+
+	gHUD.AddHudElem(this);
+	return 0;
+}
+
+int CHudCustomTimer::VidInit()
+{
+	m_iFlags &= ~HUD_ACTIVE;
+	m_iTimerPos = gHUD.m_scrinfo.iCharHeight * 4;
+
+	return 1;
+}
+
+int CHudCustomTimer::Draw(float time)
+{
+	if (gHUD.m_flTime >= m_flTurnoffTime  || gHUD.m_iIntermission) {
+		m_iFlags &= ~HUD_ACTIVE;
+		gEngfuncs.pfnPlaySoundByName("fvox/bell.wav", 1);
+		return 0;
+	}
+
+	char str[64];
+	int r, g, b;
+	UnpackRGB(r, g, b, gHUD.m_iDefaultHUDColor);
+
+	FillRGBA((ScreenWidth / 2) - 50, m_iTimerPos - 8, (-(gHUD.m_flTime - m_flTurnoffTime) / m_flSeconds) * 100, 4, r, g, b, 210);
```

Comment:
This doesn't overlap any other HUD elements, right?

Reply:
It doesn't at the moment, since there's no HUD element in the center that's using `iCharHeight * 3` in Open AG. In the original AG 6.6 there's a map name HUD element that appears after joining a server that'd overlap, but we don't have that in OpenAG.

```
AgDrawHudStringCentered(ScreenWidth / 2 , gHUD.m_scrinfo.iCharHeight*3, ScreenWidth,AgMapname().c_str(),r,g,b);
```